katsu/mueller.py

```diff
@@ -363,47 +448,91 @@ def decompose_diattenuator(M, normalize=False):
     # First, determine the diattenuator
     T = M[..., 0, 0]
 
-    if M.ndim > 2:
-        diattenuation_vector = M[..., 0, 1:] / T[..., np.newaxis]
-    else:
-        diattenuation_vector = M[..., 0, 1:] / T
+    if np.__name__ == "jax.numpy":
+        if M.ndim > 2:
+            diattenuation_vector = M.at[..., 0, 1:].divide(T.at[..., np.newaxis])
+        else:
+            diattenuation_vector = M.at[..., 0, 1:].divide(T)
 
-    D = np.sqrt(np.sum(diattenuation_vector * diattenuation_vector, axis=-1))
-    mD = np.sqrt(1 - D**2)
+        D = np.sqrt(np.sum(np.matmul(diattenuation_vector, diattenuation_vector), axis=-1))
```

Comment:
It looks like an element wise multiplication was replaced with a matmul. Does this end up doing a dot product because it does so in the last axis?

Reply:
Yeah I think this might need to be reverted to an element-wise multiplication, but let me check the source material first.

Reply:
I just tried reverting the change and uncommenting the test for it. still getting a failure unfortunately (though this could still be because of the eigen values thing)

Reply:
So decompose_diattenuator doesn’t rely on eigenvalue computation, which suggests that maybe its something significant

Reply:
i’ll try poke around on my machine

Reply:
<img width="1145" alt="image" src="https://github.com/user-attachments/assets/0c14153f-8242-4075-ae7f-451692db8d06">
works with the version you have with matmul

Reply:
<img width="1145" alt="image" src="https://github.com/user-attachments/assets/a29cc1b7-5ef8-4fce-9767-b6245c3bc14a">
Works for me with elementwise multiplication in `decompose_diattenuator`

Reply:
is that with the test uncommented?

Reply:
If I did it right I think i just commited the change

Reply:
okay, let me push uncommenting the test and see if it passes on the workflow

Reply:
Yeah for decompose_diattenuator, not for decompose_depolarizer

Reply:
wait we still have it not required so it won't check it regardless. can you test on your computer if the tests work now?

Reply:
decompose_depolarizer fails if I uncomment it